common/database.cpp

```diff
@@ -813,292 +744,275 @@ uint32 Database::GetCharacterID(const char *name) {
-	{
-		LoadVariables(); // refresh cache
+	if (v) {
+		value = *v;
 		return true;
 	}
 
-	query = StringFormat("Insert Into variables (varname, value) values ('%s', '%s')", escaped_name.c_str(), escaped_value.c_str());
-	results = QueryDatabase(query);
-
-	if (results.RowsAffected() != 1)
-		return false;
-
-	LoadVariables(); // refresh cache
-	return true;
+	return false;
 }
 
-void Database::SetAccountCRCField(uint32 account_id, std::string field_name, uint64 checksum)
+bool Database::SetVariable(const std::string& name, const std::string& value)
 {
-	QueryDatabase(
+	auto l = VariablesRepository::GetWhere(
+		*this,
 		fmt::format(
-			"UPDATE `account` SET `{}` = '{}' WHERE `id` = {}",
-			field_name,
-			checksum,
-			account_id
+			"`name` = '{}'",
+			Strings::Escape(name)
 		)
 	);
-}
 
-bool Database::GetZoneGraveyard(const uint32 graveyard_id, uint32* graveyard_zoneid, float* graveyard_x, float* graveyard_y, float* graveyard_z, float* graveyard_heading) {
-
-	std::string query = StringFormat("SELECT zone_id, x, y, z, heading FROM graveyard WHERE id=%i", graveyard_id);
-	auto results = QueryDatabase(query);
-
-	if (!results.Success()){
+	if (l.empty()) {
 		return false;
 	}
 
-	if (results.RowCount() != 1)
-		return false;
+	auto variable = l.front();
 
-	auto row = results.begin();
+	variable.value = value;
 
-	if(graveyard_zoneid != nullptr)
-		*graveyard_zoneid = Strings::ToUnsignedInt(row[0]);
-	if(graveyard_x != nullptr)
-		*graveyard_x = Strings::ToFloat(row[1]);
-	if(graveyard_y != nullptr)
-		*graveyard_y = Strings::ToFloat(row[2]);
-	if(graveyard_z != nullptr)
-		*graveyard_z = Strings::ToFloat(row[3]);
-	if(graveyard_heading != nullptr)
-		*graveyard_heading = Strings::ToFloat(row[4]);
+	const int updated = VariablesRepository::UpdateOne(*this, variable);
 
-	return true;
-}
+	if (updated) {
+		LoadVariables();
 
-uint8 Database::GetPEQZone(uint32 zone_id, uint32 version){
+		return true;
+	}
 
-	auto z = GetZoneVersionWithFallback(zone_id, version);
+	auto e = VariablesRepository::NewEntity();
 
-	return z ? z->peqzone : 0;
-}
+	e.varname = name;
+	e.value   = value;
 
-bool Database::CheckNameFilter(std::string name, bool surname)
-{
-	name = Strings::ToLower(name);
+	e = VariablesRepository::InsertOne(*this, e);
 
-	// the minimum 4 is enforced by the client too
-	if (name.empty() || name.size() < 4) {
+	if (!e.id) {
+		return false;
+	}
+
+	LoadVariables();
+
+	return true;
+}
+
+void Database::SetAccountCRCField(uint32 account_id, const std::string& field_name, uint64 checksum)
+{
+	auto e = AccountRepository::FindOne(*this, account_id);
+
+	if (Strings::EqualFold(field_name, "crc_basedata")) {
+		e.crc_basedata = checksum;
+	} else if (Strings::EqualFold(field_name, "crc_eqgame")) {
+		e.crc_eqgame = checksum;
+	} else if (Strings::EqualFold(field_name, "crc_skillcaps")) {
+		e.crc_skillcaps = checksum;
+	}
+
+	AccountRepository::UpdateOne(*this, e);
+}
+
+bool Database::GetZoneGraveyard(
```

Comment:
This function is a terrible access pattern, passing pointers and returning their values. This function is only called once so lets remove it. 

Do the repository call where this function is being called and just use it inline in the code

```cpp
bool GraveYardLoaded = content_db.GetZoneGraveyard(graveyard_id(), &pgraveyard_zoneid, &m_graveyard.x, &m_graveyard.y, &m_graveyard.z, &m_graveyard.w);
```